### PR TITLE
Group validation errors by type

### PR DESCRIPTION
**Pull request by Bjwebb**
#1117

### CODE REVIEW

**Comment by Bjwebb** [comment]
This depends on https://github.com/OpenDataServices/lib-cove/pull/9 and https://github.com/OpenDataServices/lib-cove-web/pull/11

**Comment by BibianaC** [comment]
@Bjwebb Travis failed.

**Comment by Bjwebb** [comment]
Looks like its the bulk secret data tests that are failing.

As discussed on standup, I think we can merge this without the bulk secret data tests passing (as they're a bit of a pain to update, and it's mostly as good to do them all in one go).

File: cove_360/lib/threesixtygiving.py [hunk 1]
```diff
@@ -108,10 +109,24 @@ def common_checks_360(context, upload_dir, json_data, schema_obj):
     common_checks = common_checks_context(upload_dir, json_data, schema_obj, schema_name, context)
     cell_source_map = common_checks['cell_source_map']
 
+    validation_errors = context['validation_errors']
+    validation_errors_grouped = defaultdict(list)
```
Comment (BibianaC) [inline cove_360/lib/threesixtygiving.py]: Would it not be better to extract this to a new function and call it in this one?

File: cove_360/templates/cove_360/explore.html [hunk 2]
```diff
@@ -149,17 +149,65 @@ <h3 class="panel-title panel-title-explore">
...
+              {% include "validation_table.html" %}
+            </div>
+          </div>
+          {% for error_json, values in validation_errors %}
+            {% with error=error_json|json_decode %}
+              {% cove_modal_errors className="validation-errors-"|concat:error_prefix|concat:forloop.counter modalTitle=error.message errorList=values file_type=file_type full_table=True %}
+            {% endwith %}
+          {% endfor %}
+          {% endif %}
+          {% endwith %}
```
Comment (BibianaC) [inline cove_360/templates/cove_360/explore.html]: Can these three blocks be indented? it is difficult to see where they come from if not. Same for below blocks.

Comment (Bjwebb) [reply cove_360/templates/cove_360/explore.html]: Agreed, done.

File: cove_360/lib/threesixtygiving.py (hunk 1 above)
Comment (Bjwebb) [reply cove_360/lib/threesixtygiving.py]: Yes!

**Comment by BibianaC** [comment]
Thanks for the changes @Bjwebb

**Pull request merged**